charm/templates/snap-build-poller_cron.j2

```diff
@@ -1,3 +1,4 @@
 # DO NOT EDIT: Managed by juju charm
-42 * * * * {{user}} cd {{code_dir}} && KNEX_CONFIG_PATH={{knex_config_path}} npm run poll-repositories >> {{logs_dir}}/poller.log 2>&1
+SHELL=/bin/bash
```

Comment:
This wouldn't be necessary if you just changed `source secrets.env` to `. secrets.env` below.  As far as I can see there are no other bashisms.

Reply:
oh, thanks. Fixed.